src/redturtle/prenotazioni/utilities/dateutils.py

```diff
@@ -5,9 +5,28 @@
 
 import six
 from plone.app.event.base import default_timezone
+from plone.memoize import forever
 
 from redturtle.prenotazioni import tznow
 
+# Born to be monkeypatched for the tests
+TIMEZONE_CACHE = True
+
+
+# NOTE: If the site timezone was changed, you need to reload the instance due to forever.memoize usage \cc @mamico
+def get_default_timezone(as_tzinfo):
+    @forever.memoize
+    def cached_call():
+        return default_timezone(as_tzinfo=as_tzinfo)
+
+    def uncached_call():
+        return default_timezone(as_tzinfo=as_tzinfo)
+
+    if TIMEZONE_CACHE:
+        return cached_call()
+    else:
+        return uncached_call()
+
```

Comment:
```suggestion
    if TIMEZONE_CACHE:
        return forever.memoize(default_timezone)(as_tzinfo=as_tzinfo)
    else:
        return default_timezone(as_tzinfo=as_tzinfo)
```

se il decoratore memoize viene chiamato su una funzione senza parametri non usa come chiave i parametri e avresti stessi valori in cache per paramtri differenti